torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1941,23 +2025,48 @@ def evaluate_expr(self, expr: "sympy.Expr", hint=None):
                     ShapeGuard(sympy.Eq(expr, concrete_val), stack))  # type: ignore[arg-type]
         return concrete_val
 
-def _should_allocate(user_marked_dynamic, assume_static_by_default):
-    """
-    Mainly here for readability, repurposes the flag name for the context
-    of shape_env, which cares about allocation.
-    """
-    if user_marked_dynamic:
-        return True
-    # If we got here, the user did *NOT* mark this dim as dynamic,
-    # but BC behavior is to allocate a symbol anyway.
-    return not assume_static_by_default
+class MinMaxConstraint:
```

Comment:
Why doesn't this just hold a ValueRange directly

Reply:
Hmm, it started off a little different and basically ended up the same. Can refactor.

Reply:
Yes please